src/torch/SIRF_torch.py

```diff
@@ -0,0 +1,123 @@
+try:
+    import torch
+except ImportError:
+    raise ImportError('Failed to import torch. Please install PyTorch first.')
+
+# based on 
+# https://github.com/educating-dip/pet_deep_image_prior/blob/main/src/deep_image_prior/torch_wrapper.py
+
+
+class _objectiveFunctionModule3D(torch.autograd.Function):
+    @staticmethod
+    def forward( ctx, x, image_template, sirf_obj):
+        ctx.device = x.device
+        ctx.sirf_obj = sirf_obj
+        ctx.image_template = image_template
+        ctx.x = x.detach().cpu().numpy().squeeze()
+        ctx.x = ctx.image_template.fill(ctx.x)
+        value_np = ctx.sirf_obj.get_value(ctx.x)
+        return torch.tensor(value_np).to(ctx.device)
+
+    @staticmethod
+    def backward(
+            ctx, 
+            in_grad):
+        grads_np = ctx.sirf_obj.get_gradient(ctx.x).as_array()
+        grads = torch.from_numpy(grads_np).to(ctx.device) * in_grad
+        return grads.unsqueeze(dim=0), None, None, None
+
+class _AcquisitionModelForward(torch.autograd.Function):
```

Comment:
Why `Forward`? this should be the default. If there isn't a torch function to swap forward/backward, then we could have `BackwardAcquisitionModel` I guess